Remove mesh_modifier tests and cleanup their descendants' tests

socratesgorilla opened the pull request:
Closes #22267

Commit:
Remove mesh_modifier tests #22267

moosebuild commented:
Job [Modules recover](https://civet.inl.gov/job/1189335/) on 80b9953 : invalidated by @milljm

Commit:
Replaced a symbolic link of a kitten to an actual picture of a kitten

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1189559/) on b79b2b5 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>ad10c4</th>
      <th colspan=4>#22268 b79b2b</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>84.29%</td>
      <td>84.27%</td>
      <td>-0.02%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>80084</td>
      <td>80063</td>
      <td>-21</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>14921</td>
      <td>14942</td>
      <td>+21</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/22268/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/22268/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/richards)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/22268/coverage/xfem)

</details>



This comment will be updated on new commits.

Commit:
Removed unused tests, adjusted existing tests to replace lost code coverage

Commit:
Fixed cli_args for extra_nodeset exception test

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1190110/) on 7a241d8 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/22268/site)

This comment will be updated on new commits.

socratesgorilla commented:
Remaining code coverage differences are all for lines that are covered only when running --distributed-mesh, so these are in fact covered by the distributed-mesh tests we run on Framework, but for whatever reason don't show up on the coverage report.

GiudGiud commented:
@socratesgorilla I already opened a PR for that. 
The issue is coverage went down, so there are some tests missing when you remove the mesh modifiers
https://github.com/idaholab/moose/pull/20845

You're welcome to help on my PR, this PR is behind on the work that needs to be done though

Commit:
Added/modified tests to keep coverage the same, removed unneeded golds

Commit:
Corrected error in parallel test spec

Commit:
Corrected errors in test spec

Commit:
Changed block_deletion_test5 to use block_id for last line of coverage

socratesgorilla commented:
All code coverage issues gone, ready for merge. I think there are strong reasons to prefer this over the other PR as well.

GiudGiud commented:
nice. I m happy to close the other one if we have all that was there. 
@lindsayad who was reviewing the other one or myself will look at this one.

lindsayad commented:
> All code coverage issues gone, ready for merge. I think there are strong reasons to prefer this over the other PR as well.

For a potential reviewer, can you outline the strong reasons?

socratesgorilla commented:
There

> > All code coverage issues gone, ready for merge. I think there are strong reasons to prefer this over the other PR as well.
> 
> For a potential reviewer, can you outline the strong reasons?

The other PR keeps around a lot of tests that probably don't need to be there. For example, there are ~15 block_deletion_generator tests that all perform the same operation on slightly different meshes and the other PR keeps them, whereas here anything that does not seem to test important functionalities has simply been outright removed. If a test exists and removing it doesn't lose coverage, and aren't, for example, exception tests, then it has been removed. The result is this PR is much leaner than the previous one.

lindsayad commented on test/tests/meshgenerators/block_deletion_generator/block_deletion_test5.i:
```diff
@@ -26,7 +26,7 @@
   [./ed0]
     type = BlockDeletionGenerator
-    block = 1
+    block_id = 1
```
Did you make this change in order to preserve coverage? 

I would rename this test since you deleted tests 2, 3, and 4

socratesgorilla replied:
Yep, the coverage was lost on the block_id deprecated param, and yeah, the test can be renamed

GiudGiud commented:
Coverage does not show the whole picture.
These tests were likely designed to test a capability, not a line of code. Capabilities are much more important to test than lines of codes. What do these detail= fields say?

Unless you show that all those capabilities are still tested by other tests, those tests need to stay.

socratesgorilla commented:
> Coverage does not show the whole picture. These tests were likely designed to test a capability, not a line of code. Capabilities are much more important to test than lines of codes. What do these detail= fields say?
> 
> Unless you show that all those capabilities are still tested by other tests, those tests need to stay.

Give me a few minutes and I'll go through each test and compile a list on everything

Commit:
Renamed block_deletion tests, moved some remaining tests

socratesgorilla commented:
Discussed a few of these with cody, if necessary we should all have a call/discussion about the merits of both sides. Regardless, here is the list of all the files I have moved/deleted and why:

- add_all_sidesets, add_side_sets, add_side_sets_from_bounding_box, assign_subdomain_id, break_boundary, break_mesh_by_blocks, boundingbox_nodeset, parsed_sideset, parsed_subdomain,  sidesets_between_subdomains, smooth_mesh, subdomain_bounding_box, transform: 
	All these folders are duplicates of existing tests. All have been fully deleted.

- add_extra_nodeset:
	Added exception test for bad coordinates to extra_nodeset_generator. Only meaningful difference between this folder's tests and those in extra_nodeset_generator is the mesh_modifier version uses a FileMesh for its input over a GeneratedMesh. Removed.

- assign_element_subdomain_id:
	1. quad_with_subdomainid_test.i has functionality tested by "sidesets_between_subdomains_generator/sideset_between_vector_subdomains_generator.i", "userobjects/internal_side_user_object/internal_side_user_object_two_materials.i", and both tests in transfers/multiapp_transfer_transformation, all of which feature a QUAD4.
	2. quad_with_elementid_subdomainid_test.i has functionality tested in postprocessors/point_value/point_value.i
	3. For the remaining test, talked about it with cody, added in latest push.

- block_deleter:
	Here are the details of the non-duplicate tests in this folder:

      2: detail = "a 3D concave subdomain;"

      3: detail = "a 2D interior subdomain;"

      4: detail = "a 3D interior subdomain;"

      5: detail = "a 2D non-concave subdomain;"

      6: detail = "a 3D non-concave subdomain;"

      7: detail = "a 2D removal of a union of disjoint pieces;"

      8: detail = "a 2D removal of a subdomain containing a nodeset;"

      9: detail = "a 2D removal of a subdomain that eliminates sideset;"

      10: detail = "a 2D removal of a subdomain containing a sideset;"

      12: detail = "a 2D concave subdomain with a cut across elements."

    The problem with all of these tests is why they might seem like useful tests, and perhaps are, they aren't testing anything meaningful about the *block deletion*. ElementDeletionGeneratorBase does not have any logic for doing anything specific for these cases, it merely deletes the elems with no granularity. What is actually being tested here is the contract() and prepare_for_use() functions in libMesh, which is where those tests should be located if they are necessary. They aren't testing any particular functionality of the block deletion and I feel removing them is valid. The only one that has been kept was the one testing the deprecated param and parallel aspects for code coverage.
   

- image_subdomain:
	1. image_2d.i has functionality tested in preconditioners/fsp/fsp_test_image.i
	2. image_3d.i is a duplicate of functions/image_function/image_3d.i

- lower_d_block:
	1. The first_order test is a duplicate of ids.i in lower_d_block_generator
	2. The second_order test can be found in many of the mortar tests, but just for an example, mortar/aux-gap/mismatch.i captures the same functionality.

- mesh_extruder:
	All duplicates except tri, which was moved.

- modifier_depend_order
	1. depend_with_prepare.i was testing a removed param and has no other meaningful tests
	2. modifier_depend_order.i seems to absolutely be a weirdly converted mesh_modifier test and TransformGenerators are tested plenty in mesh_generators/transform_generator and in things like cyclic.i. Removed.

- sidesets_around_subdomain:
	Duplicates, but parallel coverage was lost and replaced.

Commit:
Fixed documentation pointing to a renamed file
